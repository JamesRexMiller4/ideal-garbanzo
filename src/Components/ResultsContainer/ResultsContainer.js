import React, { useState } from 'react';
import './ResultsContainer.scss';
import ResultCard from '../ResultCard/ResultCard';
import next from '../../icons/next.svg';
import leftArrow from '../../icons/left-arrow.svg';

const ResultsContainer = ({results}) => {
  const [ page, setPage ] = useState(1);
  const TOTALRESULTSLENGTH = Math.round(results.length / 10);

  const paginateResults = (resultData) => {
    const start = (page - 1) * 10;
    const end = start + 10;

    return resultData.slice(start, end);
  };

  const updateCurrentPage = (currentPage, incrementor) => {
    if (incrementor < 0) {
      currentPage === 1 ? setPage(1) : setPage(page - 1);
    }
    if (incrementor > 0) {
      currentPage === TOTALRESULTSLENGTH ? setPage(TOTALRESULTSLENGTH)
      : setPage(page + 1);
    }
  };

  const resultCards = paginateResults(results).map(result => (
  <ResultCard key={result.id} id={result.it} result={result}/>))
  
  return ( 
    <section className="results-container-section">
      <div className='results-pagination-div'>
        <img id="left-arrow" onClick={() => updateCurrentPage(page, -1)} src={leftArrow} alt="previous-results"/>
        <h2>{"Showing " + page + " of " + (Math.round(results.length / 10))}</h2>
        <img id="right-arrow" onClick={() => updateCurrentPage(page, 1)} src={next} alt="more-results"/>
      </div>
        {resultCards}
    </section>
  );
}

export default ResultsContainer;