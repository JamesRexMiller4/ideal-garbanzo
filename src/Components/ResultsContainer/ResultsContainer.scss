@import '../../scss/variables.scss';

.results-container-section {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: lighten($base-blue, 80);
  border-top: 1px solid;
  box-shadow: 8px 4px 2px -4px black;
}