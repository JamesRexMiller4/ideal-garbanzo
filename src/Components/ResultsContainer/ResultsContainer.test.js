import React from 'react';
import {shallow} from 'enzyme';
import ResultsContainer from './ResultsContainer';
import mockData from '../../data/mockData';
 
describe('ResultsContainer', () => {
  let wrapper, largerMockData;
 
  beforeEach(() => {
    largerMockData = [...mockData, ...mockData, ...mockData]
    wrapper = shallow(<ResultsContainer results={largerMockData}/>)
  })
 
  it('should render the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
 
  it('should update the snapshot when an arrow is clicked', () => {
    wrapper.find("#right-arrow").simulate('click');
    expect(wrapper).toMatchSnapshot();
  });
});