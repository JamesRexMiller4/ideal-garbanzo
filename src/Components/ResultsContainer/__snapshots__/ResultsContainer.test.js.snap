// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsContainer should render the snapshot 1`] = `
<section
  className="results-container-section"
>
  <div
    className="results-pagination-div"
  >
    <img
      alt="previous-results"
      id="left-arrow"
      onClick={[Function]}
      src="left-arrow.svg"
    />
    <h2>
      Showing 1 of 2
    </h2>
    <img
      alt="more-results"
      id="right-arrow"
      onClick={[Function]}
      src="next.svg"
    />
  </div>
  <ResultCard
    key="f223fdd0-4adc-423e-9747-980a66c256ca"
    result={
      Object {
        "address1": "201 Waterfront St",
        "attire": "business casual",
        "city": "Oxon Hill",
        "genre": "Steak,American,Contemporary,Seafood,Cafe",
        "hours": "Open Daily 5:30 PM-10:00 PM",
        "id": "f223fdd0-4adc-423e-9747-980a66c256ca",
        "lat": "38.782098",
        "long": "-77.017492",
        "name": "Old Hickory Steakhouse",
        "state": "MD",
        "tags": "Social,Food and Dining,Restaurants,Steakhouses",
        "telephone": "(301) 965-4000",
        "website": "http://www.gaylordnational.com",
        "zip": "20745",
      }
    }
  />
  <ResultCard
    key="00b35e1a-82b1-4988-b8b9-6df826db2818"
    result={
      Object {
        "address1": "303 E Main St",
        "attire": "business casual",
        "city": "Aspen",
        "genre": "Japanese,Sushi,Asian,Contemporary,Seafood",
        "hours": "Open Daily 5:30 PM-9:00 PM",
        "id": "00b35e1a-82b1-4988-b8b9-6df826db2818",
        "lat": "39.190723",
        "long": "-106.82031",
        "name": "Matsuhisa",
        "state": "CO",
        "tags": "Social,Food and Dining,Restaurants,Japanese,Social,Food and Dining,Restaurants,Sushi",
        "telephone": "(970) 544-6628",
        "website": "http://www.matsuhisaaspen.com",
        "zip": "81611",
      }
    }
  />
  <ResultCard
    key="0f41a3d0-0641-4eef-b7fd-706f083cf0d5"
    result={
      Object {
        "address1": "108 3rd St NE",
        "attire": "business casual",
        "city": "Charlottesville",
        "genre": "French,European,Cafe,Continental,American",
        "hours": "Mon-Thu 5:30 PM-9:00 PM; Fri-Sat 5:30 PM-10:00 PM",
        "id": "0f41a3d0-0641-4eef-b7fd-706f083cf0d5",
        "lat": "38.030526",
        "long": "-78.479785",
        "name": "Fleurie Restaurant",
        "state": "VA",
        "tags": "Social,Food and Dining,Restaurants,French",
        "telephone": "(434) 971-7800",
        "website": "http://www.fleurierestaurant.com",
        "zip": "22902",
      }
    }
  />
  <ResultCard
    key="0b4bfe46-3e60-4de4-82ba-2dd8e5d46b56"
    result={
      Object {
        "address1": "500 Crescent Ct",
        "attire": "business casual",
        "city": "Dallas",
        "genre": "Steak,American",
        "hours": "Mon-Thu 11:00 AM-10:00 PM; Fri 11:00 AM-11:00 PM; Sat 5:00 PM-11:00 PM; Sun 5:00 PM-9:00 PM",
        "id": "0b4bfe46-3e60-4de4-82ba-2dd8e5d46b56",
        "lat": "32.794749",
        "long": "-96.804099",
        "name": "The Capital Grille",
        "state": "TX",
        "tags": "Social,Food and Dining,Restaurants,Steakhouses,Social,Food and Dining,Restaurants,American",
        "telephone": "(214) 303-0500",
        "website": "http://www.thecapitalgrille.com",
        "zip": "75201",
      }
    }
  />
  <ResultCard
    key="cd273a24-f8de-44f6-8add-028e22229293"
    result={
      Object {
        "address1": "8731 International Dr",
        "attire": "Casual",
        "city": "Orlando",
        "genre": "Seafood,International,American,Oysters,Cafe",
        "hours": "Mon-Fri 4:00 PM-10:00 PM; Sat-Sun 2:00 PM-10:00 PM",
        "id": "cd273a24-f8de-44f6-8add-028e22229293",
        "lat": "28.43897",
        "long": "-81.470707",
        "name": "Boston Lobster Feast",
        "state": "FL",
        "tags": "Social,Food and Dining,Restaurants,Seafood,Social,Food and Dining,Restaurants,American",
        "telephone": "(407) 248-8606",
        "website": "http://www.bostonlobsterfeast.com",
        "zip": "32819",
      }
    }
  />
  <ResultCard
    key="f223fdd0-4adc-423e-9747-980a66c256ca"
    result={
      Object {
        "address1": "201 Waterfront St",
        "attire": "business casual",
        "city": "Oxon Hill",
        "genre": "Steak,American,Contemporary,Seafood,Cafe",
        "hours": "Open Daily 5:30 PM-10:00 PM",
        "id": "f223fdd0-4adc-423e-9747-980a66c256ca",
        "lat": "38.782098",
        "long": "-77.017492",
        "name": "Old Hickory Steakhouse",
        "state": "MD",
        "tags": "Social,Food and Dining,Restaurants,Steakhouses",
        "telephone": "(301) 965-4000",
        "website": "http://www.gaylordnational.com",
        "zip": "20745",
      }
    }
  />
  <ResultCard
    key="00b35e1a-82b1-4988-b8b9-6df826db2818"
    result={
      Object {
        "address1": "303 E Main St",
        "attire": "business casual",
        "city": "Aspen",
        "genre": "Japanese,Sushi,Asian,Contemporary,Seafood",
        "hours": "Open Daily 5:30 PM-9:00 PM",
        "id": "00b35e1a-82b1-4988-b8b9-6df826db2818",
        "lat": "39.190723",
        "long": "-106.82031",
        "name": "Matsuhisa",
        "state": "CO",
        "tags": "Social,Food and Dining,Restaurants,Japanese,Social,Food and Dining,Restaurants,Sushi",
        "telephone": "(970) 544-6628",
        "website": "http://www.matsuhisaaspen.com",
        "zip": "81611",
      }
    }
  />
  <ResultCard
    key="0f41a3d0-0641-4eef-b7fd-706f083cf0d5"
    result={
      Object {
        "address1": "108 3rd St NE",
        "attire": "business casual",
        "city": "Charlottesville",
        "genre": "French,European,Cafe,Continental,American",
        "hours": "Mon-Thu 5:30 PM-9:00 PM; Fri-Sat 5:30 PM-10:00 PM",
        "id": "0f41a3d0-0641-4eef-b7fd-706f083cf0d5",
        "lat": "38.030526",
        "long": "-78.479785",
        "name": "Fleurie Restaurant",
        "state": "VA",
        "tags": "Social,Food and Dining,Restaurants,French",
        "telephone": "(434) 971-7800",
        "website": "http://www.fleurierestaurant.com",
        "zip": "22902",
      }
    }
  />
  <ResultCard
    key="0b4bfe46-3e60-4de4-82ba-2dd8e5d46b56"
    result={
      Object {
        "address1": "500 Crescent Ct",
        "attire": "business casual",
        "city": "Dallas",
        "genre": "Steak,American",
        "hours": "Mon-Thu 11:00 AM-10:00 PM; Fri 11:00 AM-11:00 PM; Sat 5:00 PM-11:00 PM; Sun 5:00 PM-9:00 PM",
        "id": "0b4bfe46-3e60-4de4-82ba-2dd8e5d46b56",
        "lat": "32.794749",
        "long": "-96.804099",
        "name": "The Capital Grille",
        "state": "TX",
        "tags": "Social,Food and Dining,Restaurants,Steakhouses,Social,Food and Dining,Restaurants,American",
        "telephone": "(214) 303-0500",
        "website": "http://www.thecapitalgrille.com",
        "zip": "75201",
      }
    }
  />
  <ResultCard
    key="cd273a24-f8de-44f6-8add-028e22229293"
    result={
      Object {
        "address1": "8731 International Dr",
        "attire": "Casual",
        "city": "Orlando",
        "genre": "Seafood,International,American,Oysters,Cafe",
        "hours": "Mon-Fri 4:00 PM-10:00 PM; Sat-Sun 2:00 PM-10:00 PM",
        "id": "cd273a24-f8de-44f6-8add-028e22229293",
        "lat": "28.43897",
        "long": "-81.470707",
        "name": "Boston Lobster Feast",
        "state": "FL",
        "tags": "Social,Food and Dining,Restaurants,Seafood,Social,Food and Dining,Restaurants,American",
        "telephone": "(407) 248-8606",
        "website": "http://www.bostonlobsterfeast.com",
        "zip": "32819",
      }
    }
  />
</section>
`;

exports[`ResultsContainer should update the snapshot when an arrow is clicked 1`] = `
<section
  className="results-container-section"
>
  <div
    className="results-pagination-div"
  >
    <img
      alt="previous-results"
      id="left-arrow"
      onClick={[Function]}
      src="left-arrow.svg"
    />
    <h2>
      Showing 2 of 2
    </h2>
    <img
      alt="more-results"
      id="right-arrow"
      onClick={[Function]}
      src="next.svg"
    />
  </div>
  <ResultCard
    key="f223fdd0-4adc-423e-9747-980a66c256ca"
    result={
      Object {
        "address1": "201 Waterfront St",
        "attire": "business casual",
        "city": "Oxon Hill",
        "genre": "Steak,American,Contemporary,Seafood,Cafe",
        "hours": "Open Daily 5:30 PM-10:00 PM",
        "id": "f223fdd0-4adc-423e-9747-980a66c256ca",
        "lat": "38.782098",
        "long": "-77.017492",
        "name": "Old Hickory Steakhouse",
        "state": "MD",
        "tags": "Social,Food and Dining,Restaurants,Steakhouses",
        "telephone": "(301) 965-4000",
        "website": "http://www.gaylordnational.com",
        "zip": "20745",
      }
    }
  />
  <ResultCard
    key="00b35e1a-82b1-4988-b8b9-6df826db2818"
    result={
      Object {
        "address1": "303 E Main St",
        "attire": "business casual",
        "city": "Aspen",
        "genre": "Japanese,Sushi,Asian,Contemporary,Seafood",
        "hours": "Open Daily 5:30 PM-9:00 PM",
        "id": "00b35e1a-82b1-4988-b8b9-6df826db2818",
        "lat": "39.190723",
        "long": "-106.82031",
        "name": "Matsuhisa",
        "state": "CO",
        "tags": "Social,Food and Dining,Restaurants,Japanese,Social,Food and Dining,Restaurants,Sushi",
        "telephone": "(970) 544-6628",
        "website": "http://www.matsuhisaaspen.com",
        "zip": "81611",
      }
    }
  />
  <ResultCard
    key="0f41a3d0-0641-4eef-b7fd-706f083cf0d5"
    result={
      Object {
        "address1": "108 3rd St NE",
        "attire": "business casual",
        "city": "Charlottesville",
        "genre": "French,European,Cafe,Continental,American",
        "hours": "Mon-Thu 5:30 PM-9:00 PM; Fri-Sat 5:30 PM-10:00 PM",
        "id": "0f41a3d0-0641-4eef-b7fd-706f083cf0d5",
        "lat": "38.030526",
        "long": "-78.479785",
        "name": "Fleurie Restaurant",
        "state": "VA",
        "tags": "Social,Food and Dining,Restaurants,French",
        "telephone": "(434) 971-7800",
        "website": "http://www.fleurierestaurant.com",
        "zip": "22902",
      }
    }
  />
  <ResultCard
    key="0b4bfe46-3e60-4de4-82ba-2dd8e5d46b56"
    result={
      Object {
        "address1": "500 Crescent Ct",
        "attire": "business casual",
        "city": "Dallas",
        "genre": "Steak,American",
        "hours": "Mon-Thu 11:00 AM-10:00 PM; Fri 11:00 AM-11:00 PM; Sat 5:00 PM-11:00 PM; Sun 5:00 PM-9:00 PM",
        "id": "0b4bfe46-3e60-4de4-82ba-2dd8e5d46b56",
        "lat": "32.794749",
        "long": "-96.804099",
        "name": "The Capital Grille",
        "state": "TX",
        "tags": "Social,Food and Dining,Restaurants,Steakhouses,Social,Food and Dining,Restaurants,American",
        "telephone": "(214) 303-0500",
        "website": "http://www.thecapitalgrille.com",
        "zip": "75201",
      }
    }
  />
  <ResultCard
    key="cd273a24-f8de-44f6-8add-028e22229293"
    result={
      Object {
        "address1": "8731 International Dr",
        "attire": "Casual",
        "city": "Orlando",
        "genre": "Seafood,International,American,Oysters,Cafe",
        "hours": "Mon-Fri 4:00 PM-10:00 PM; Sat-Sun 2:00 PM-10:00 PM",
        "id": "cd273a24-f8de-44f6-8add-028e22229293",
        "lat": "28.43897",
        "long": "-81.470707",
        "name": "Boston Lobster Feast",
        "state": "FL",
        "tags": "Social,Food and Dining,Restaurants,Seafood,Social,Food and Dining,Restaurants,American",
        "telephone": "(407) 248-8606",
        "website": "http://www.bostonlobsterfeast.com",
        "zip": "32819",
      }
    }
  />
</section>
`;
